### PR TITLE
Fix issue with mmio r/w in windows

### PR DESCRIPTION
**Pull request by npmitche**
Signed-off-by: Nathaniel Mitchell <nathaniel.p.mitchell@intel.com>

### CODE REVIEW

**Review by Sae86 — APPROVED** [review]

File: drivers/win7/driver.c [hunk 1]
```diff
@@ -322,12 +322,51 @@ NTSTATUS _write_mmio_mem(PHYSICAL_ADDRESS pa, unsigned int len, void* pData)
...
-    if (len == 8) 
-      { 
-        count = 2; 
-      }
     DbgPrint("[chipsec] writing %u bytes to MMIO address 0x%08x_%08x (virtual = %#010x)", len, pa.HighPart, pa.LowPart, (UINTN)va);
-    WRITE_REGISTER_BUFFER_ULONG((volatile ULONG*)(va), (PULONG)pData, count);
+    
+    switch(len)
+		{
+			case 1:
```
Comment (ErikBjorge) [inline drivers/win7/driver.c]: Looks like we have some TAB characters here.  Can we clean that up?

File: drivers/win7/driver.c [hunk 2]
```diff
@@ -322,12 +322,51 @@ NTSTATUS _write_mmio_mem(PHYSICAL_ADDRESS pa, unsigned int len, void* pData)
...
+    switch(len)
+    {
+      case 1:
+        WRITE_REGISTER_BUFFER_UCHAR((volatile UCHAR*)(va), (PUCHAR)pData, count);
+        break;
+      case 2:
+        WRITE_REGISTER_BUFFER_USHORT((volatile USHORT*)(va), (PUSHORT)pData, count);
+        break;
+      case 8:
+        count = 2;
```
Comment (ErikBjorge) [inline drivers/win7/driver.c:336]: Add a comment here that the missing break is intentional and that it just sets up two DWORD writes.

File: drivers/win7/driver.c [hunk 3]
```diff
@@ -322,12 +322,51 @@ NTSTATUS _write_mmio_mem(PHYSICAL_ADDRESS pa, unsigned int len, void* pData)
...
+    switch(len)
+    {
+      case 1:
+        READ_REGISTER_BUFFER_UCHAR((volatile UCHAR*)(va), (PUCHAR)pData, count);
+        break;
+      case 2:
+        READ_REGISTER_BUFFER_USHORT((volatile USHORT*)(va), (PUSHORT)pData, count);
+        break;
+      case 8:
+        count = 2;
```
Comment (ErikBjorge) [inline drivers/win7/driver.c:365]: Comment same as in write.

**Comment by npmitche** [comment]
_only change from 6cdc15c to current was somehow some tabs found their way in, I replaced them with spaces_

File: drivers/win7/driver.c [hunk 4]
```diff
@@ -872,8 +953,8 @@ DriverDeviceControl(
             // --
             // -- read MSR to check if it was written
             // --
-//            _rdmsr( _msr_addr, &_eax, &_edx );
-//            DbgPrint( "[chipsec][IOCTL_WRMSR] RDMSR( 0x%x ) --> 0x%08x%08x\n", _msr_addr, _edx, _eax );
+          //  _rdmsr( _msr_addr, &_eax, &_edx );
+          //  DbgPrint( "[chipsec][IOCTL_WRMSR] RDMSR( 0x%x ) --> 0x%08x%08x\n", _msr_addr, _edx, _eax );
```
Comment (ErikBjorge) [inline drivers/win7/driver.c:957]: Why are we maintaining dead code?  I would have been fine with just a removal or no change at all.

**Comment by npmitche** [comment]
Fixed: https://github.com/chipsec/chipsec/compare/6cdc15c..dda69e65534f6d6c9db891078c9bbab0e7c0ed7a

**Review by ErikBjorge — APPROVED** [review]

File: chipsec/helper/win/win32helper.py [hunk 5]
```diff
@@ -553,8 +554,9 @@ def native_write_phys_mem( self, phys_address_hi, phys_address_lo, length, buf )
     # @TODO: Temporarily the same as read_phys_mem for compatibility
     def read_mmio_reg( self, phys_address, size ):
         out_size = size
+        logger().log("size: {} addr: {}".format(size, phys_address))
```
Comment (BrentHoltsclaw) [inline chipsec/helper/win/win32helper.py:557]: Should this always log?

File: chipsec/helper/win/win32helper.py [hunk 6]
```diff
@@ -553,8 +554,9 @@ def native_write_phys_mem( self, phys_address_hi, phys_address_lo, length, buf )
     # @TODO: Temporarily the same as read_phys_mem for compatibility
```
Comment (BrentHoltsclaw) [inline chipsec/helper/win/win32helper.py:554]: Remove comment?

File: chipsec/helper/win/win32helper.py (hunk 5 above)
Comment (npmitche) [reply chipsec/helper/win/win32helper.py:557]: Good catch. I thought I fixed all of the logging.

File: chipsec/helper/win/win32helper.py (hunk 6 above)
Comment (npmitche) [reply chipsec/helper/win/win32helper.py:554]: probably... XD

**Pull request merged**